In [2]:
import pandas as pd

A continuacion realizo la lectura de los csv

In [3]:
penetracion_p100hab = pd.read_csv('./DatasetsObligatorios/Penetración-poblacion.csv')
penetracion_pcia = pd.read_csv('./DatasetsObligatorios/Penetracion-hogares.csv')
accesos_baf = pd.read_csv('./DatasetsObligatorios/Totales Dial-BAf.csv')
accesos_baf_pcia = pd.read_csv('./DatasetsObligatorios/Dial-BAf.csv')
accesos_tec = pd.read_csv('./DatasetsObligatorios/Accesos Por Tecnología.csv')
accesos_vmd = pd.read_csv('./DatasetsObligatorios/Totales VMD.csv')
accesos_vel_prov = pd.read_csv('./DatasetsObligatorios/Accesos por velocidad.csv')
accesos_totales = pd.read_csv('./DatasetsObligatorios/Totales Accesos por velocidad.csv')
accesos_prov_vel = pd.read_csv('./DatasetsObligatorios/Velocidad_sin_Rangos.csv')
ingresos = pd.read_csv('./DatasetsObligatorios/Ingresos .csv')
accesos_loc = pd.read_csv('./DatasetsObligatorios/Acc_vel_loc_sinrangos.csv')
acceso_tec_loc = pd.read_csv('./DatasetsObligatorios/Accesos_tecnologia_localidad.csv')
mapa_conect = pd.read_csv('./DatasetsObligatorios/mapa_conectividad.csv')
metadata = pd.read_csv('./DatasetsComplementarios/Metadata_mundial.csv')
mundial = pd.read_csv('./DatasetsComplementarios/mundial.csv')


---

Comienzo a analizar los diferentes Datos de los csv para realizar un proceso de ETL, para su posterior EDA, y su final presentacion en un Dashboard.
* El objetivo de este analisis profundo es minimizar la cantidad de datasets y poseer los datos correctos y necesarios para su presentacion final en PowerBi.

In [4]:
penetracion_p100hab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       984 non-null    int64  
 1   Trimestre                 984 non-null    int64  
 2   Provincia                 984 non-null    object 
 3   Accesos por cada 100 hab  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [5]:
penetracion_pcia.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int64  
 1   Trimestre                     984 non-null    int64  
 2   Provincia                     984 non-null    object 
 3   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [6]:
#Combino columnas año, Trimestre, Provincia
df1_comb = pd.merge(penetracion_p100hab, penetracion_pcia, on =['Año', 'Trimestre','Provincia'])
df1_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int64  
 1   Trimestre                     984 non-null    int64  
 2   Provincia                     984 non-null    object 
 3   Accesos por cada 100 hab      984 non-null    float64
 4   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(2), int64(2), object(1)
memory usage: 38.6+ KB


In [7]:
df1_comb.sample(2)

Año  Trimestre  Provincia  Accesos por cada 100 hab  \
338  2020          3  Catamarca                 11.995289   
128  2022          4    Formosa                 10.030000   

     Accesos por cada 100 hogares  
338                     47.157099  
128                     39.410000

---------------

In [8]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 986 entries, 0 to 985
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           985 non-null    object 
 1   Trimestre     985 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), object(3)
memory usage: 69.5+ KB


In [9]:
accesos_tec.head(5)

Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2024         1     Buenos Aires  203297.0   2790663.0     1799013.0   
1  2024         1  Capital Federal   70108.0   1177475.0      203756.0   
2  2024         1        Catamarca    5901.0     11025.0       56542.0   
3  2024         1            Chaco   11343.0     58929.0       64463.0   
4  2024         1           Chubut   26306.0     77217.0       26356.0   

   Wireless     Otros      Total  
0  150290.0  101294.0  5044557.0  
1    4493.0   48879.0  1504711.0  
2    1384.0      81.0    74933.0  
3    8404.0   12404.0   155543.0  
4   30867.0    9930.0   170676.0

In [10]:
accesos_tec.isna().sum()

Año             1
Trimestre       1
Provincia       2
ADSL            2
Cablemodem      2
Fibra óptica    2
Wireless        2
Otros           2
Total           2
dtype: int64

In [11]:
accesos_tec = accesos_tec.dropna()

In [12]:
#visualizo el Dataframe y encuentro '*' en Año y Trimestre. Reemplazo el *
accesos_tec = accesos_tec.copy()
accesos_tec['Año'].unique() 
accesos_tec['Año'] = accesos_tec['Año'].str.replace('[^0-9]', '').str.rstrip('*')
accesos_tec['Año'] = accesos_tec['Año'].astype(int)
#access_tec.head(5)
accesos_tec['Año'].unique() 

array([2024, 2023, 2022, 2021, 2020, 2019, 2018, 2017, 2016, 2015, 2014])

In [13]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
Index: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    int32  
 1   Trimestre     984 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), int32(1), object(2)
memory usage: 73.0+ KB


In [14]:
accesos_tec['Trimestre'] = accesos_tec['Trimestre'].str.replace('[^0-9]', '').str.rstrip('*')
accesos_tec['Trimestre'] = accesos_tec['Trimestre'].astype(int)

In [15]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
Index: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    int32  
 1   Trimestre     984 non-null    int32  
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), int32(2), object(1)
memory usage: 69.2+ KB


In [16]:
# Combino el df previo creado con las mismas columnas
df2_comb = pd.merge(accesos_tec, df1_comb, on=['Año', 'Trimestre', 'Provincia'])
df2_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    float64
 4   Cablemodem                    984 non-null    float64
 5   Fibra óptica                  984 non-null    float64
 6   Wireless                      984 non-null    float64
 7   Otros                         984 non-null    float64
 8   Total                         984 non-null    float64
 9   Accesos por cada 100 hab      984 non-null    float64
 10  Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(8), int32(2), object(1)
memory usage: 77.0+ KB


In [17]:
df2_comb.sample(5)

Año  Trimestre   Provincia     ADSL  Cablemodem  Fibra óptica  Wireless  \
712  2016          4       Salta  79389.0     26657.0         110.0     158.0   
830  2015          3     Neuquén  47102.0     36620.0        1045.0   13051.0   
834  2015          3    San Luis  11432.0      1025.0         354.0      31.0   
811  2015          4  Santa Cruz  15614.0      2723.0         396.0    5261.0   
681  2017          1       Jujuy  33869.0      9278.0          22.0       0.0   

      Otros     Total  Accesos por cada 100 hab  Accesos por cada 100 hogares  
712   621.0  106935.0                  7.865488                     32.694120  
830  2798.0  100616.0                 16.195213                     54.599315  
834   377.0   13219.0                  2.768812                      9.706503  
811  2409.0   26403.0                  8.143243                     27.744769  
681    34.0   43203.0                  5.819769                     22.756986

In [18]:
# Paso todos los datos a interger
df2_comb = df2_comb.astype({
    'Provincia': 'string',
    'ADSL': 'int',
    'Cablemodem': 'int',
    'Wireless': 'int',
    'Otros': 'int',
    'Total': 'int',
    'Fibra óptica': 'int',
    'Accesos por cada 100 hab': 'int',
    'Accesos por cada 100 hogares': 'int'
})



In [19]:
df2_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Año                           984 non-null    int32 
 1   Trimestre                     984 non-null    int32 
 2   Provincia                     984 non-null    string
 3   ADSL                          984 non-null    int32 
 4   Cablemodem                    984 non-null    int32 
 5   Fibra óptica                  984 non-null    int32 
 6   Wireless                      984 non-null    int32 
 7   Otros                         984 non-null    int32 
 8   Total                         984 non-null    int32 
 9   Accesos por cada 100 hab      984 non-null    int32 
 10  Accesos por cada 100 hogares  984 non-null    int32 
dtypes: int32(10), string(1)
memory usage: 46.3 KB


In [20]:
#Unifico DF con el mismo tipo de informacion.
df3_comb = pd.merge(df2_comb, accesos_baf_pcia, on=['Año', 'Trimestre', 'Provincia'])
df3_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 14 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total_x                       984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    int32  
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Total

---

In [21]:
accesos_totales.sample(3)

Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
31  2016          2           44008                   384221   
5   2022          4           31981                    85668   
12  2021          1           39487                    41674   

    Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
31                5058481                  796998                   785759   
5                 1075272                 1109421                   765015   
12                2593477                 1072722                   737930   

    Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS     Total  
31                    73977           29020       0   7172464  
5                    335573         7410902  394282  11208114  
12                   595920         4379965  186797   9647972

In [22]:
df4_comb = pd.merge(df3_comb, accesos_totales, on=['Año','Trimestre'])


In [23]:
df4_comb.sample(3)

Año  Trimestre Provincia   ADSL  Cablemodem  Fibra óptica  Wireless  \
20   2024          1  Santa Fe  86548      399817        374288     44162   
322  2020          4  La Pampa   9009       58945         15432     12194   
306  2021          1  San Luis   8417       10678            28     63826   

     Otros  Total_x  Accesos por cada 100 hab  ...  Total_y  Hasta 512 kbps  \
20   13773   918588                        25  ...   918588           28801   
322   1315    96895                        26  ...    96895           39510   
306  36205   119154                        23  ...   119154           39487   

     Entre 512 Kbps y 1 Mbps  Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
20                     69355                 866152                  950930   
322                    42185                2622638                 1073875   
306                    41674                2593477                 1072722   

     Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS  \
20                    672155                   353896         8363694  328173   
322                   786595                   582420         4239237  185102   
306                   737930                   595920         4379965  186797   

        Total  
20   11633156  
322   9571562  
306   9647972  

[3 rows x 23 columns]

---

In [24]:
df_nac = pd.merge(accesos_baf, accesos_vmd, on=['Año', 'Trimestre'])
df_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     41 non-null     int64  
 1   Trimestre               41 non-null     int64  
 2   Banda ancha fija        41 non-null     int64  
 3   Dial up                 41 non-null     int64  
 4   Total                   41 non-null     int64  
 5   Periodo                 41 non-null     object 
 6   Mbps (Media de bajada)  41 non-null     float64
 7   Trimestre.1             41 non-null     object 
dtypes: float64(1), int64(5), object(2)
memory usage: 2.7+ KB


In [25]:
df2_nac = pd.merge(df_nac, accesos_vel_prov, on =['Año', 'Trimestre'])
df3_nac = pd.merge(df2_nac, ingresos, on =['Año', 'Trimestre'])
df4_nac = pd.merge(df3_nac, accesos_tec, on =['Año', 'Trimestre'])

In [26]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Periodo_x                  22464 non-null  object 
 6   Mbps (Media de bajada)     22464 non-null  float64
 7   Trimestre.1                22464 non-null  object 
 8   Provincia_x                22464 non-null  object 
 9   HASTA 512 kbps             22464 non-null  float64
 10  + 512 Kbps - 1 Mbps        22464 non-null  float64
 11  + 1 Mbps - 6 Mbps          22464 non-null  float64
 12  + 6 Mbps - 10 Mbps         22464 non-null  float64
 13  + 10 Mbps - 20 Mbps        22464 non-null  flo

In [27]:
df_partidos = pd.merge(acceso_tec_loc, mapa_conect, on = ['Provincia', 'Partido', 'Localidad'])
df_partidos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2826 entries, 0 to 2825
Data columns (total 26 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       2826 non-null   object 
 1   Partido         2826 non-null   object 
 2   Localidad       2826 non-null   object 
 3   Link Indec      2826 non-null   object 
 4   ADSL_x          2826 non-null   int64  
 5   CABLEMODEM      2826 non-null   int64  
 6   DIAL UP         2826 non-null   int64  
 7   FIBRA OPTICA    2826 non-null   int64  
 8   OTROS           2826 non-null   int64  
 9   SATELITAL       2826 non-null   int64  
 10  WIMAX           2826 non-null   int64  
 11  WIRELESS        2826 non-null   int64  
 12  Total general   2826 non-null   int64  
 13  Población       2826 non-null   int64  
 14  ADSL_y          2826 non-null   object 
 15  Cablemódem      2826 non-null   object 
 16  Dial Up         2826 non-null   object 
 17  Fibra óptica    2826 non-null   o

In [28]:
df2_partidos = pd.merge(df_partidos, accesos_loc, on = ['Provincia', 'Partido', 'Localidad'])
df2_partidos.head(5)

Provincia     Partido               Localidad Link Indec_x  ADSL_x  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo      6854100     755   
1  BUENOS AIRES  25 de Mayo               Del Valle      6854020     181   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte      6854040       0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra      6854060       0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde      6854050       0   

   CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  ...  680 Mbps  \
0        4600        0             2      0        742  ...       NaN   
1           0        0             0     10          1  ...       NaN   
2           0        0             0      0          0  ...       NaN   
3         782        0           496     16          0  ...       NaN   
4           0        0             0      0          0  ...       NaN   

   700 Mbps  800 Mbps  850 Mbps 900 Mbps 999 Mbps 1000 Mbps 1024 Mbps  \
0       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
1       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
2       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
3       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
4       NaN       NaN       NaN      NaN      NaN       NaN       NaN   

  4000 Mbps 10000 Mbps  
0       NaN        NaN  
1       NaN        NaN  
2       NaN        NaN  
3       NaN        NaN  
4       NaN        NaN  

[5 rows x 148 columns]

Duplicados:

In [29]:
# Contar filas duplicadas
numero_duplicados = df2_partidos.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 0


In [30]:
# Elimino las filas duplicadas
df4_nac = df4_nac.drop_duplicates()

# Verifico el número de filas después de eliminar las duplicadas
print(f"El número de filas después de eliminar duplicados: {df4_nac.shape[0]}")


El número de filas después de eliminar duplicados: 22464


---

In [31]:
# Evaluo los resultados del trabajo
igual_1 = (df4_comb['Total'] == df4_comb['Total_x']).sum()
igual_2 = (df4_comb['Total'] == df4_comb['Total_y']).sum()
igual_3 = (df4_comb['Total_x'] == df4_comb['Total_y']).sum()

print(igual_1)
print(igual_2)
print(igual_3)

0
0
984


Visualizo que no existen coincidencias entre 'Total' y 'Total_x', tampoco entre 'Total' y 'Total_y'; pero si hay 984 coincidencias entre 'Total_x' y 'Total_y', se elimina la columna.


In [32]:
df4_comb = df4_comb.drop('Total_y', axis=1)
df4_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 22 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total_x                       984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    int32  
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Hasta

In [33]:
# Renombro algunas columnas necesarias. 
df4_comb = df4_comb.rename(columns = {'Total': 'Total suma Mbps'})
df4_comb = df4_comb.rename(columns = {'Total_x': 'Total conexion'})
df4_comb = df4_comb.rename(columns = {'OTROS': 'Otros Mbps'})
df4_comb = df4_comb.rename(columns = {'HASTA 512 kbps': 'Hasta 512 kbps'})

In [34]:
df4_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 22 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total conexion                984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    int32  
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Hasta

In [35]:
df_group = df4_comb.groupby(['Año', 'Trimestre', 'Provincia']).sum()
df_group.sample(5)

ADSL  Cablemodem  Fibra óptica  Wireless  Otros  \
Año  Trimestre Provincia                                                       
2017 2         Chaco        54449       35613           458        52     39   
2021 2         Río Negro    48189       43761         18727     15031  18376   
2016 3         Entre Ríos  112588       61850           206      6441     64   
2023 3         Salta        29113       98324         59914     12337   6545   
2015 4         Chaco        56818       29456           186        52     41   

                           Total conexion  Accesos por cada 100 hab  \
Año  Trimestre Provincia                                              
2017 2         Chaco                90611                         7   
2021 2         Río Negro           144084                        19   
2016 3         Entre Ríos          181149                        13   
2023 3         Salta               206233                        13   
2015 4         Chaco                86553                         7   

                           Accesos por cada 100 hogares  Banda ancha fija  \
Año  Trimestre Provincia                                                    
2017 2         Chaco                                 29             90597   
2021 2         Río Negro                             61            143583   
2016 3         Entre Ríos                            45            181092   
2023 3         Salta                                 57            206228   
2015 4         Chaco                                 28             86537   

                           Dial up  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
Año  Trimestre Provincia                                                      
2017 2         Chaco          14.0           22366                   222766   
2021 2         Río Negro     501.0           40172                    42024   
2016 3         Entre Ríos     57.0           20104                   294413   
2023 3         Salta           5.0           30274                    80609   
2015 4         Chaco          16.0           34243                   427394   

                           Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
Año  Trimestre Provincia                                                   
2017 2         Chaco                     4416065                  990281   
2021 2         Río Negro                 2531271                 1080279   
2016 3         Entre Ríos                4816081                  936533   
2023 3         Salta                      945608                  998725   
2015 4         Chaco                     5049640                  726740   

                           Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  \
Año  Trimestre Provincia                                                      
2017 2         Chaco                       1342000                   231604   
2021 2         Río Negro                    693277                   647401   
2016 3         Entre Ríos                   919208                   124564   
2023 3         Salta                        720084                   350918   
2015 4         Chaco                        639011                    17568   

                           Más de 30 Mbps  Otros Mbps  Total suma Mbps  
Año  Trimestre Provincia                                                
2017 2         Chaco                98635           0          7323717  
2021 2         Río Negro          4661291      167369          9863084  
2016 3         Entre Ríos           31022           0          7141925  
2023 3         Salta              8003269      277421         11406908  
2015 4         Chaco                22170           0          6916766

Duplicados:


In [36]:
# Contar filas duplicadas
numero_duplicados = df_group.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 0


-----

In [37]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Periodo_x                  22464 non-null  object 
 6   Mbps (Media de bajada)     22464 non-null  float64
 7   Trimestre.1                22464 non-null  object 
 8   Provincia_x                22464 non-null  object 
 9   HASTA 512 kbps             22464 non-null  float64
 10  + 512 Kbps - 1 Mbps        22464 non-null  float64
 11  + 1 Mbps - 6 Mbps          22464 non-null  float64
 12  + 6 Mbps - 10 Mbps         22464 non-null  float64
 13  + 10 Mbps - 20 Mbps        22464 non-null  flo

In [38]:
df4_nac = df4_nac.rename(columns = {'HASTA 512 kbps': 'Hasta 512 kbps'})

In [39]:
df4_nac = df4_nac.drop(['Periodo_y','Periodo_x'], axis = 1)
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 25 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Mbps (Media de bajada)     22464 non-null  float64
 6   Trimestre.1                22464 non-null  object 
 7   Provincia_x                22464 non-null  object 
 8   Hasta 512 kbps             22464 non-null  float64
 9   + 512 Kbps - 1 Mbps        22464 non-null  float64
 10  + 1 Mbps - 6 Mbps          22464 non-null  float64
 11  + 6 Mbps - 10 Mbps         22464 non-null  float64
 12  + 10 Mbps - 20 Mbps        22464 non-null  float64
 13  + 20 Mbps - 30 Mbps        22464 non-null  flo

In [40]:
df4_nac

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
0      2024          1          11621363    11793  11633156   
1      2024          1          11621363    11793  11633156   
2      2024          1          11621363    11793  11633156   
3      2024          1          11621363    11793  11633156   
4      2024          1          11621363    11793  11633156   
...     ...        ...               ...      ...       ...   
22459  2014          1           6362108    36290   6398398   
22460  2014          1           6362108    36290   6398398   
22461  2014          1           6362108    36290   6398398   
22462  2014          1           6362108    36290   6398398   
22463  2014          1           6362108    36290   6398398   

       Mbps (Media de bajada)   Trimestre.1   Provincia_x  Hasta 512 kbps  \
0                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
1                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
2                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
3                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
4                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
...                       ...           ...           ...             ...   
22459                3.617127  Ene-Mar 2014       Tucumán             6.0   
22460                3.617127  Ene-Mar 2014       Tucumán             6.0   
22461                3.617127  Ene-Mar 2014       Tucumán             6.0   
22462                3.617127  Ene-Mar 2014       Tucumán             6.0   
22463                3.617127  Ene-Mar 2014       Tucumán             6.0   

       + 512 Kbps - 1 Mbps  ...     OTROS    Total_y  \
0                  22510.0  ...  125147.0  5044557.0   
1                  22510.0  ...  125147.0  5044557.0   
2                  22510.0  ...  125147.0  5044557.0   
3                  22510.0  ...  125147.0  5044557.0   
4                  22510.0  ...  125147.0  5044557.0   
...                    ...  ...       ...        ...   
22459              34672.0  ...       0.0   130032.0   
22460              34672.0  ...       0.0   130032.0   
22461              34672.0  ...       0.0   130032.0   
22462              34672.0  ...       0.0   130032.0   
22463              34672.0  ...       0.0   130032.0   

       Ingresos (miles de pesos)          Provincia_y      ADSL  Cablemodem  \
0                   2.804191e+08         Buenos Aires  203297.0   2790663.0   
1                   2.804191e+08      Capital Federal   70108.0   1177475.0   
2                   2.804191e+08            Catamarca    5901.0     11025.0   
3                   2.804191e+08                Chaco   11343.0     58929.0   
4                   2.804191e+08               Chubut   26306.0     77217.0   
...                          ...                  ...       ...         ...   
22459               2.984054e+06           Santa Cruz   18927.0       100.0   
22460               2.984054e+06             Santa Fe  322662.0    174296.0   
22461               2.984054e+06  Santiago Del Estero   32567.0      3598.0   
22462               2.984054e+06     Tierra Del Fuego   21618.0      2837.0   
22463               2.984054e+06              Tucumán  129717.0        83.0   

       Fibra óptica  Wireless     Otros      Total  
0         1799013.0  150290.0  101294.0  5044557.0  
1          203756.0    4493.0   48879.0  1504711.0  
2           56542.0    1384.0      81.0    74933.0  
3           64463.0    8404.0   12404.0   155543.0  
4           26356.0   30867.0    9930.0   170676.0  
...             ...       ...       ...        ...  
22459         381.0    4092.0    3264.0    26764.0  
22460        3059.0    5951.0     644.0   506612.0  
22461          19.0     915.0      34.0    37133.0  
22462         648.0       1.0    2934.0    28038.0  
22463         121.0      13.0      98.0   130032.0  

[22464 rows x 25 columns]

In [41]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 25 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Mbps (Media de bajada)     22464 non-null  float64
 6   Trimestre.1                22464 non-null  object 
 7   Provincia_x                22464 non-null  object 
 8   Hasta 512 kbps             22464 non-null  float64
 9   + 512 Kbps - 1 Mbps        22464 non-null  float64
 10  + 1 Mbps - 6 Mbps          22464 non-null  float64
 11  + 6 Mbps - 10 Mbps         22464 non-null  float64
 12  + 10 Mbps - 20 Mbps        22464 non-null  float64
 13  + 20 Mbps - 30 Mbps        22464 non-null  flo

In [42]:
#Revision de Iguales
igual_4 = (df4_nac['Total_x'] == df4_nac['Total_y']).sum()
print(igual_4)

0


In [43]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
16449  2016          3           7178340    28610   7206950   
745    2023          3          11395115    11793  11406908   
18369  2015          4           6952289    32542   6984831   
1020   2023          3          11395115    11793  11406908   
16480  2016          3           7178340    28610   7206950   

       Mbps (Media de bajada)    Trimestre.1          Provincia_x  \
16449                5.938777  Jul-Sept 2016             Misiones   
745                129.670000  Jul-Sept 2023           Entre Ríos   
18369                4.990449   Oct-Dic 2015  Santiago Del Estero   
1020               129.670000  Jul-Sept 2023             San Luis   
16480                5.938777  Jul-Sept 2016              Neuquén   

       Hasta 512 kbps  + 512 Kbps - 1 Mbps  ...   OTROS   Total_y  \
16449          1010.0               6030.0  ...     0.0   97998.0   
745             107.0               4510.0  ...     0.0  293293.0   
18369           703.0               7836.0  ...     0.0   39138.0   
1020             60.0                628.0  ...  8484.0  148916.0   
16480          1531.0               1527.0  ...     0.0  106026.0   

       Ingresos (miles de pesos)      Provincia_y     ADSL  Cablemodem  \
16449               6.912442e+06            Jujuy  34250.0      8236.0   
745                 1.331066e+08  Capital Federal  60765.0   1205339.0   
18369               5.376899e+06            Jujuy  32934.0      6933.0   
1020                1.331066e+08          Mendoza  17677.0     73066.0   
16480               6.912442e+06            Salta  80866.0     26113.0   

       Fibra óptica  Wireless    Otros      Total  
16449          22.0       0.0     30.0    42538.0  
745        167428.0    4629.0  29823.0  1467984.0  
18369          22.0       0.0     35.0    39924.0  
1020       178719.0   27609.0    951.0   298022.0  
16480          90.0     157.0    565.0   107791.0  

[5 rows x 25 columns]

In [44]:
#Elimino los '.'
df4_nac['Hasta 512 kbps'] = df4_nac['Hasta 512 kbps'].astype(str).str.replace('.',',')

In [45]:
# Creo una Funcion para reemplazar los '+' y los '-'
def renombre_col(columna):
    columna = columna.replace('+', 'Entre')
    columna = columna.replace('-','y')
    return columna

In [46]:
df4_nac.rename(columns=renombre_col, inplace=True)

In [47]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
20661  2014          4           6559172    39324   6598496   
2813   2022          3          11078691    12437  11091128   
18548  2015          3           6902267    32801   6935068   
4063   2021          4          10476933    12861  10489794   
20578  2014          4           6559172    39324   6598496   

       Mbps (Media de bajada)    Trimestre.1          Provincia_x  \
20661                4.156888   Oct-Dic 2014             Santa Fe   
2813                62.460000  Jul-Sept 2022  Santiago Del Estero   
18548                4.788911  Jul-Sept 2015               Chubut   
4063                52.340000   Oct-Dic 2021      Capital Federal   
20578                4.156888   Oct-Dic 2014             San Juan   

      Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  ...  OTROS    Total_y  \
20661         8707,0                  99962.0  ...    0.0   527471.0   
2813           178,0                   1384.0  ...  792.0   119032.0   
18548          641,0                   5886.0  ...    0.0    78362.0   
4063           563,0                   1695.0  ...    0.0  1412119.0   
20578           11,0                    478.0  ...    0.0    52119.0   

       Ingresos (miles de pesos)          Provincia_y      ADSL  Cablemodem  \
20661               3.950441e+06  Santiago Del Estero   34110.0      4231.0   
2813                6.408678e+07              Córdoba  163770.0    465461.0   
18548               5.153739e+06             Santa Fe  337042.0    209673.0   
4063                4.546789e+07           Entre Ríos   70459.0    133387.0   
20578               3.950441e+06             La Pampa   16043.0     34281.0   

       Fibra óptica  Wireless    Otros      Total  
20661          19.0     920.0     37.0    39317.0  
2813       338570.0   61070.0   9797.0  1038668.0  
18548        4169.0    4426.0   1428.0   556738.0  
4063        23452.0   25022.0  12726.0   265046.0  
20578         514.0    5894.0    648.0    57380.0  

[5 rows x 25 columns]

In [48]:
# Redondeo los numeros 
df4_nac = df4_nac.round(2)

In [49]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
3618   2022          1          10611390    12619  10624009   
20107  2015          1           6699714    38018   6737732   
13584  2017          4           7842778    27444   7870222   
958    2023          3          11395115    11793  11406908   
1508   2023          2          11239781    12116  11251897   

       Mbps (Media de bajada)    Trimestre.1          Provincia_x  \
3618                    55.11   Ene-Mar 2022           Corrientes   
20107                    4.35   Ene-Mar 2015  Santiago Del Estero   
13584                   12.01   Oct-Dic 2017              Neuquén   
958                    129.67  Jul-Sept 2023            Río Negro   
1508                   123.95   Abr-Jun 2023              Neuquén   

      Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  ...    OTROS   Total_y  \
3618            67,0                   4029.0  ...   7094.0  142578.0   
20107          701,0                   9236.0  ...      0.0   38617.0   
13584         1396,0                   2372.0  ...   6105.0  115843.0   
958            112,0                   1061.0  ...  14957.0  163485.0   
1508           168,0                    984.0  ...   2969.0  161746.0   

       Ingresos (miles de pesos)       Provincia_y       ADSL  Cablemodem  \
3618                5.521931e+07          San Luis     7987.0     11296.0   
20107               4.876385e+06        Santa Cruz    16215.0      2700.0   
13584               1.122613e+07      Buenos Aires  1241172.0   1718013.0   
958                 1.331066e+08  Tierra Del Fuego     9357.0     45203.0   
1508                1.180603e+08          Santa Fe   126344.0    399669.0   

       Fibra óptica  Wireless    Otros      Total  
3618           31.0   71809.0  28721.0   119844.0  
20107         392.0    4772.0   2562.0    26641.0  
13584      131181.0   60082.0  31448.0  3181896.0  
958          3682.0     157.0    332.0    58731.0  
1508       311166.0   44134.0  14501.0   895814.0  

[5 rows x 25 columns]

In [50]:
#Elimino las columnas redundantes
df4_nac = df4_nac.drop(['Total_y','Total_x'], axis = 1)
df4_nac

Año  Trimestre  Banda ancha fija  Dial up  Mbps (Media de bajada)  \
0      2024          1          11621363    11793                  139.15   
1      2024          1          11621363    11793                  139.15   
2      2024          1          11621363    11793                  139.15   
3      2024          1          11621363    11793                  139.15   
4      2024          1          11621363    11793                  139.15   
...     ...        ...               ...      ...                     ...   
22459  2014          1           6362108    36290                    3.62   
22460  2014          1           6362108    36290                    3.62   
22461  2014          1           6362108    36290                    3.62   
22462  2014          1           6362108    36290                    3.62   
22463  2014          1           6362108    36290                    3.62   

        Trimestre.1   Provincia_x Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
0      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
1      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
2      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
3      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
4      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
...             ...           ...            ...                      ...   
22459  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22460  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22461  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22462  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22463  Ene-Mar 2014       Tucumán            6,0                  34672.0   

       Entre 1 Mbps y 6 Mbps  ...  Entre 30 Mbps     OTROS  \
0                   221185.0  ...      4068292.0  125147.0   
1                   221185.0  ...      4068292.0  125147.0   
2                   221185.0  ...      4068292.0  125147.0   
3                   221185.0  ...      4068292.0  125147.0   
4                   221185.0  ...      4068292.0  125147.0   
...                      ...  ...            ...       ...   
22459                83210.0  ...            0.0       0.0   
22460                83210.0  ...            0.0       0.0   
22461                83210.0  ...            0.0       0.0   
22462                83210.0  ...            0.0       0.0   
22463                83210.0  ...            0.0       0.0   

       Ingresos (miles de pesos)          Provincia_y      ADSL  Cablemodem  \
0                   2.804191e+08         Buenos Aires  203297.0   2790663.0   
1                   2.804191e+08      Capital Federal   70108.0   1177475.0   
2                   2.804191e+08            Catamarca    5901.0     11025.0   
3                   2.804191e+08                Chaco   11343.0     58929.0   
4                   2.804191e+08               Chubut   26306.0     77217.0   
...                          ...                  ...       ...         ...   
22459               2.984054e+06           Santa Cruz   18927.0       100.0   
22460               2.984054e+06             Santa Fe  322662.0    174296.0   
22461               2.984054e+06  Santiago Del Estero   32567.0      3598.0   
22462               2.984054e+06     Tierra Del Fuego   21618.0      2837.0   
22463               2.984054e+06              Tucumán  129717.0        83.0   

      Fibra óptica  Wireless     Otros      Total  
0        1799013.0  150290.0  101294.0  5044557.0  
1         203756.0    4493.0   48879.0  1504711.0  
2          56542.0    1384.0      81.0    74933.0  
3          64463.0    8404.0   12404.0   155543.0  
4          26356.0   30867.0    9930.0   170676.0  
...            ...       ...       ...        ...  
22459        381.0    4092.0    3264.0    26764.0  
22460       3059.0    5951.0     644.0   506612.

In [51]:
#Renombramos para un mejor entendimiento.
df4_nac = df4_nac.rename(columns = {'OTROS': 'Otros Mbps'})
df4_nac = df4_nac.rename(columns = {'Otros': 'Otras conexiones'})
df4_nac = df4_nac.rename(columns = {'Total': 'Total Mbps-conexiones'})
df4_nac = df4_nac.rename(columns = {'Entre 30 Mbps': 'Mas de 30 Mbps'})
df4_nac = df4_nac.rename(columns = {'Provincia_y': 'Provincia'})

In [52]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Mbps (Media de bajada)     22464 non-null  float64
 5   Trimestre.1                22464 non-null  object 
 6   Provincia_x                22464 non-null  object 
 7   Hasta 512 kbps             22464 non-null  object 
 8   Entre 512 Kbps y 1 Mbps    22464 non-null  float64
 9   Entre 1 Mbps y 6 Mbps      22464 non-null  float64
 10  Entre 6 Mbps y 10 Mbps     22464 non-null  float64
 11  Entre 10 Mbps y 20 Mbps    22464 non-null  float64
 12  Entre 20 Mbps y 30 Mbps    22464 non-null  float64
 13  Mas de 30 Mbps             22464 non-null  flo

In [53]:
columns_to_convert = [
    'Ingresos (miles de pesos)', 
    'ADSL', 
    'Cablemodem', 
    'Fibra óptica', 
    'Wireless', 
    'Otras conexiones', 
    'Total Mbps-conexiones'
]

df4_nac[columns_to_convert] = df4_nac[columns_to_convert].astype(int)

In [54]:
# cambio los'.' por ','
df4_nac['Hasta 512 kbps'] = df4_nac['Hasta 512 kbps'].str.replace(',', '.')

In [55]:
print(df4_nac['Entre 512 Kbps y 1 Mbps'].dtype)


float64


In [56]:
df4_nac['Entre 512 Kbps y 1 Mbps'] = df4_nac['Entre 512 Kbps y 1 Mbps'].astype(str)


In [57]:
df4_nac['Entre 512 Kbps y 1 Mbps'] = df4_nac['Entre 512 Kbps y 1 Mbps'].str.replace(',', '.')

In [58]:
a_float = ['Hasta 512 kbps',
    'Entre 512 Kbps y 1 Mbps']
df4_nac[a_float] = df4_nac[a_float].astype(float)

In [59]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Mbps (Media de bajada)     22464 non-null  float64
 5   Trimestre.1                22464 non-null  object 
 6   Provincia_x                22464 non-null  object 
 7   Hasta 512 kbps             22464 non-null  float64
 8   Entre 512 Kbps y 1 Mbps    22464 non-null  float64
 9   Entre 1 Mbps y 6 Mbps      22464 non-null  float64
 10  Entre 6 Mbps y 10 Mbps     22464 non-null  float64
 11  Entre 10 Mbps y 20 Mbps    22464 non-null  float64
 12  Entre 20 Mbps y 30 Mbps    22464 non-null  float64
 13  Mas de 30 Mbps             22464 non-null  flo

Duplicados:

In [60]:
# Contar filas duplicadas
numero_duplicados = df4_nac.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")


Número de filas duplicadas: 0


-----

In [61]:
df2_partidos.info(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2804 entries, 0 to 2803
Data columns (total 148 columns):
 #    Column          Dtype  
---   ------          -----  
 0    Provincia       object 
 1    Partido         object 
 2    Localidad       object 
 3    Link Indec_x    object 
 4    ADSL_x          int64  
 5    CABLEMODEM      int64  
 6    DIAL UP         int64  
 7    FIBRA OPTICA    int64  
 8    OTROS           int64  
 9    SATELITAL       int64  
 10   WIMAX           int64  
 11   WIRELESS        int64  
 12   Total general   int64  
 13   Población       int64  
 14   ADSL_y          object 
 15   Cablemódem      object 
 16   Dial Up         object 
 17   Fibra óptica    object 
 18   Satelital       object 
 19   Wireless        object 
 20   Telefonía Fija  object 
 21   3G              object 
 22   4G              object 
 23   Link            float64
 24   Latitud         float64
 25   Longitud        float64
 26   Link Indec_y    object 
 27   Otros           

In [62]:
df2_partidos = df2_partidos.merge(mapa_conect[['Partido', 'Latitud', 'Longitud']], on='Partido', how='left')


In [63]:
df2_partidos.sample(5)

Provincia                  Partido        Localidad Link Indec_x  \
923    BUENOS AIRES               Avellaneda   Villa Domínico      6035010   
7896   BUENOS AIRES                San Pedro        San Pedro      6770050   
28574         JUJUY                San Pedro      La Mendieta     38063090   
50544      SANTA FE                San Justo  Colonia Dolores     82112030   
39132         SALTA  Grl. José de San Martín           Pacará     66056170   

       ADSL_x  CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  ...  \
923        15        8462        2          7525     19          0  ...   
7896     1260        5176        0         10123     78          3  ...   
28574       0           0        0             0      0          1  ...   
50544       0           0        0             0      0          0  ...   
39132       0           0        0             0      2          0  ...   

       800 Mbps  850 Mbps  900 Mbps  999 Mbps 1000 Mbps 1024 Mbps 4000 Mbps  \
923         NaN       NaN       NaN       NaN     672.0       NaN       NaN   
7896        NaN       NaN       NaN       NaN      23.0       NaN       NaN   
28574       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
50544       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
39132       NaN       NaN       NaN       NaN       NaN       NaN       NaN   

      10000 Mbps  Latitud_y Longitud_y  
923          NaN -34.662651 -58.365699  
7896         NaN -33.769590 -59.638371  
28574        NaN -26.619790 -54.114682  
50544        NaN -30.348496 -60.521785  
39132        NaN -23.258187 -63.339056  

[5 rows x 150 columns]

In [64]:
df2_partidos.dtypes

Provincia        object
Partido          object
Localidad        object
Link Indec_x     object
ADSL_x            int64
                 ...   
1024 Mbps       float64
4000 Mbps       float64
10000 Mbps      float64
Latitud_y       float64
Longitud_y      float64
Length: 150, dtype: object

In [65]:
# Renombro Columnas
nuevo_nombre = {'ADSL_x': 'ADSL',
                'FIBRA OPTICA': 'Fibra Optica',
                'OTROS': 'Otras Conexiones',
                'CABLEMODEM': 'Cable Modem',
                'SATELITAL': 'Satelital',
                'Telefonía Fija': 'Telefonia Fija',
                'Latitud_y': 'Latitud',
                'Longitud_y': 'Longitud'}

In [66]:
df2_partidos = df2_partidos.rename(columns = nuevo_nombre)

In [67]:
#Visualizo
df2_partidos.head(5)

Provincia     Partido   Localidad Link Indec_x  ADSL  Cable Modem  \
0  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
1  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
2  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
3  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
4  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   

   DIAL UP  Fibra Optica  Otras Conexiones  Satelital  ...  800 Mbps  \
0        0             2                 0        742  ...       NaN   
1        0             2                 0        742  ...       NaN   
2        0             2                 0        742  ...       NaN   
3        0             2                 0        742  ...       NaN   
4        0             2                 0        742  ...       NaN   

   850 Mbps  900 Mbps  999 Mbps 1000 Mbps 1024 Mbps 4000 Mbps 10000 Mbps  \
0       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
1       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
2       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
3       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
4       NaN       NaN       NaN       NaN       NaN       NaN        NaN   

     Latitud   Longitud  
0 -35.433939 -60.173121  
1 -35.739217 -60.559063  
2 -35.897345 -60.731609  
3 -35.270234 -59.557503  
4 -35.164397 -60.081358  

[5 rows x 150 columns]

Duplicados:

In [68]:
# Contar filas duplicadas
numero_duplicados = df2_partidos.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 600


In [69]:
# Elimino filas duplicadas
df4_nac = df4_nac.drop_duplicates()

# Verifico el número de filas después de eliminar duplicados
print(f"Número de filas después de eliminar duplicados: {df4_nac.shape[0]}")


Número de filas después de eliminar duplicados: 22464


----

Visualizo los DataFrames resultantes

In [70]:
df4_comb.sample(2)

Año  Trimestre Provincia   ADSL  Cablemodem  Fibra óptica  Wireless  \
388  2020          1    Chubut  45763       27287          1324     16885   
945  2014          2     Jujuy  30128        4161            23        40   

     Otros  Total conexion  Accesos por cada 100 hab  ...  Dial up  \
388   5479           96738                        15  ...    800.0   
945      9           34361                         4  ...      2.0   

     Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  Entre 1 Mbps y 6 Mbps  \
388           42550                    56170                2649819   
945           41064                   656408                5149574   

     Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  \
388                 1022014                   814470                   532309   
945                  341689                   147273                      478   

     Más de 30 Mbps  Otros Mbps  Total suma Mbps  
388         3500882      194212          8812426  
945           12259           0          6348745  

[2 rows x 22 columns]

In [71]:
df_group.sample(2)

ADSL  Cablemodem  Fibra óptica  Wireless  Otros  \
Año  Trimestre Provincia                                                       
2014 2         Santa Cruz   18956         448           381      4118   3266   
2016 1         Córdoba     350884      252346           674      5636    386   

                           Total conexion  Accesos por cada 100 hab  \
Año  Trimestre Provincia                                              
2014 2         Santa Cruz           27169                         8   
2016 1         Córdoba             609926                        16   

                           Accesos por cada 100 hogares  Banda ancha fija  \
Año  Trimestre Provincia                                                    
2014 2         Santa Cruz                            29             26710   
2016 1         Córdoba                               55            609692   

                           Dial up  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
Año  Trimestre Provincia                                                      
2014 2         Santa Cruz    459.0           41064                   656408   
2016 1         Córdoba       234.0           30428                   404810   

                           Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
Año  Trimestre Provincia                                                   
2014 2         Santa Cruz                5149574                  341689   
2016 1         Córdoba                   4944358                  762999   

                           Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  \
Año  Trimestre Provincia                                                      
2014 2         Santa Cruz                   147273                      478   
2016 1         Córdoba                      641646                    27664   

                           Más de 30 Mbps  Otros Mbps  Total suma Mbps  
Año  Trimestre Provincia                                                
2014 2         Santa Cruz           12259           0          6348745  
2016 1         Córdoba              23380           0          6835285

In [72]:
df_nac.sample(2)

Año  Trimestre  Banda ancha fija  Dial up    Total        Periodo  \
13  2020          4           9561546    10016  9571562   Oct-Dic 2020   
26  2017          3           7696594    28349  7724943  Jul-Sept 2017   

    Mbps (Media de bajada)    Trimestre.1  
13               42.360000   Oct-Dic 2020  
26                8.339048  Jul-Sept 2017

In [73]:
df2_partidos.sample(2)

Provincia             Partido Localidad Link Indec_x  ADSL  \
2764  BUENOS AIRES  General Pueyrredón     Batán      6357070  1313   
7867  BUENOS AIRES           San Pedro  Río Tala      6770040     0   

      Cable Modem  DIAL UP  Fibra Optica  Otras Conexiones  Satelital  ...  \
2764            0        0          1514                 0          1  ...   
7867            0        0             2               237          0  ...   

      800 Mbps  850 Mbps  900 Mbps  999 Mbps 1000 Mbps 1024 Mbps 4000 Mbps  \
2764       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
7867       NaN       NaN       NaN       NaN       NaN       NaN       NaN   

     10000 Mbps    Latitud   Longitud  
2764        NaN -38.236931 -57.763418  
7867        NaN -33.879456 -59.875328  

[2 rows x 150 columns]

----

Datasets finales luego del ETL:

In [74]:
#  Guardo los DataFrames resultantes
df4_comb.to_csv('df_combinado.csv',)
df_group.to_csv('df_agrupado.csv')
df4_nac.to_csv('df_nac.csv')
df2_partidos.to_csv('df_partido.csv')                                                                                                    

---

Visualizo el 5to Dataset de interes:


In [75]:
df_prov = pd.read_csv('./df_provincias.csv')
df_prov.sample(2)

centroide_lat  centroide_lon  id    nombre        nombre_completo
14     -30.706927     -60.949837  82  Santa Fe  Provincia de Santa Fe
20     -23.320078     -65.764252  38     Jujuy     Provincia de Jujuy

Duplicados:

In [76]:
# Contar filas duplicadas
numero_duplicados = df_prov.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 0


In [77]:
df_prov.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24 entries, 0 to 23
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   centroide_lat    24 non-null     float64
 1   centroide_lon    24 non-null     float64
 2   id               24 non-null     int64  
 3   nombre           24 non-null     object 
 4   nombre_completo  24 non-null     object 
dtypes: float64(2), int64(1), object(2)
memory usage: 1.1+ KB


---